tensorboard/plugins/text/text_plugin.py

```diff
@@ -265,22 +270,25 @@ def _async_index_impl(self):
         'TextPlugin index_impl() thread ending after %0.3f sec', elapsed)
 
   def index_impl(self):
+    run_to_series = self._fetch_run_to_series_from_multiplexer()
     # A previous system of collecting and serving text summaries involved
     # storing the tags of text summaries within tensors.json files. See if we
     # are currently using that system. We do not want to drop support for that
     # use case.
-    run_to_series = collections.defaultdict(list)
     name = 'tensorboard_text'
     run_to_assets = self._multiplexer.PluginAssets(name)
     for run, assets in run_to_assets.items():
       if 'tensors.json' in assets:
         tensors_json = self._multiplexer.RetrievePluginAsset(
             run, name, 'tensors.json')
         tensors = json.loads(tensors_json)
-        run_to_series[run] = tensors
+        run_to_series[run] += tensors
       else:
-        run_to_series[run] = []
+        run_to_series[run] += []
+    return run_to_series
 
+  def _fetch_run_to_series_from_multiplexer(self):
+    run_to_series = collections.defaultdict(list)
     # TensorBoard is obtaining summaries related to the text plugin based on
     # SummaryMetadata stored within Value protos.
     mapping = self._multiplexer.PluginRunToTagToContent(metadata.PLUGIN_NAME)
```

Comment:
The "Augment the summaries..." comment a few lines down (grrrr github disallowing comments outside diffs) no longer applies here, it should probably be relocated - though I'm not actually sure introducing this helper is necessary?   think the main thing is just checking the multiplexer in is_active(), otherwise index_impl() should work fine as is.

If we do keep the helper, the logic can also be simplified, this is pretty much just

`return {run: list(tag_to_content.keys()) for run, tag_to_content in six.iteritems(mapping)}`

I guess you're actually using the fact that it's a defaultdict(list) up in index_impl() but I might opt to avoid that dependency, since defaultdict is a bit subtle and relying on a given dict return value to actually be a defaultdict seems a bit hazardous unless clearly documented.

Reply:
Ah, indeed! We can concisely write the expression like that. And yes, lets use a normal dict since runs with no data map to empty lists. I kept the helper since we also use it within `tags_impl` so that route is consistent with the plugin being active.

Also, removed the duplicate comment.